backend/lib/endpoints/users.js

```diff
@@ -190,6 +191,51 @@ async function routes(app) {
       return new User(userData).save();
     },
   );
+
+  app.delete("/", { preValidation: [app.authenticate] }, async (req) => {
+    const { userId } = req;
+    // Insufficient scope to delete users on Auth0
+    // expected any of: delete:users,delete:current_user
+    // Please add any of the needed scopes and uncomment the line below
+    // await Auth0.deleteUser(getCookieToken(req), userId);
+    await Promise.all([
+      User.deleteMany({ _id: userId }),
+      Comment.deleteMany({ "author.id": userId }),
+      Post.find({ "author.id": userId }, async (err, allUserPosts) => {
+        if (err) throw err;
+        for (const userPost of allUserPosts) {
+          await Promise.all([
+            Comment.deleteMany({ postId: userPost._id }),
+            userPost.delete(),
+          ]);
+        }
+      }),
```

Comment:
I think this could be changed to a single delete for all the comments instead of a for loop (opening fewer connections to the DB and avoiding that overhead). 

Instead of iterating through the `postIds` I think a single delete with the `$in` operator will do the trick, something like:
 `Comment.deleteMany({postId: {'$in': userPostIds}})`